aiden-recommender/aiden_recommender/scrapers/field_extractors.py

```diff
@@ -0,0 +1,127 @@
+from pydantic import BaseModel
+from abc import ABC, abstractmethod
+import jmespath
+from datetime import datetime
+from pydantic_core import ValidationError
+from loguru import logger
+
+
+class AbstractFieldExtractor(ABC):
+    @staticmethod
+    def select(data: dict, query: str) -> any:
+        if not query:
+            return data
+        return jmespath.search(query, data)
+
+    @abstractmethod
+    def _extract(self, data: dict) -> any:
+        pass
+
+    @abstractmethod
+    def extract(self, data: dict, field: str) -> dict:
+        pass
+
+
+class FieldExtractorBase(AbstractFieldExtractor):
+    def __init__(self, query: str, default: any = None, transform_func: callable = None) -> None:
+        self.query = query
+        self.default = default
+        self.transform_func = transform_func if transform_func else lambda x: x
+
+    def _extract(self, data: dict, query=None) -> any:
+        query = query if query is not None else self.query
+        result = self.select(data, query)
+        return self.transform_func(result) if result is not None else self.default
+
+    def extract(self, data, field):
```

Comment:
Why don't you simply return the extrated result instead of adding it to a dict ? The other application should occupy themselves with where to put it. Imagine that in the future you want to reuse this function, or maybe change where you store the results on the Parser, this function just simply give the extracted value and nothing more in my opinion.